chia/full_node/block_store.py

```diff
@@ -57,7 +57,8 @@ async def create(cls, db_wrapper: DBWrapper):
 
         await self.db.execute("CREATE INDEX IF NOT EXISTS height on block_records(height)")
 
-        await self.db.execute("CREATE INDEX IF NOT EXISTS hh on block_records(header_hash)")
+        # Do not create named index on column header_hash, because it already is auto indexed as of Primary Key (Index Name sqlite_autoindex_block_records_1)
+        #await self.db.execute("CREATE INDEX IF NOT EXISTS hh on block_records(header_hash)")
```

Comment:
will this actually create another, identical, index? that sounds a bit weird. I would have expected sqlite to just alias the existing one. Having a name can be helpful in `INDEXED BY` clauses, when optimizing sql commands.

I'm all for removing an index we don't use, and I think we should probably leave a commented out command to drop it from the DB as well, that we can enable at some point in the future. But did you confirm that this creates a redundant index?

Reply:
Yes, indexes for Primary Key columns are created internally - [check documentation](https://www.sqlite.org/fileformat2.html#intschema) - and are not just aliases.

```
-- List all Schema Objects for Table block_records, including internal objects

sqlite> select * from sqlite_master where tbl_name = 'block_records';
type|name|tbl_name|rootpage|sql
table|block_records|block_records|4|CREATE TABLE block_records(header_hash text PRIMARY KEY, prev_hash text, height bigint,block blob, sub_epoch_summary blob, is_peak tinyint, is_block tinyint)
index|sqlite_autoindex_block_records_1|block_records|5|
index|height|block_records|15|CREATE INDEX height on block_records(height)
index|peak|block_records|16|CREATE INDEX peak on block_records(is_peak)
index|hh|block_records|6812910|CREATE INDEX hh on block_records(header_hash)

-- Activate explain plan for stmt
sqlite> .eqp on

-- instead of index hh the internal index sqlite_autoindex_block_records_1 is used for query
sqlite> SELECT block from block_records WHERE header_hash='1c6e3fc954fed0bfd6d6d44bb850910e98fe48a086ef00e9fbe3ea7e84490b40';
QUERY PLAN
--SEARCH TABLE block_records USING INDEX sqlite_autoindex_block_records_1 (header_hash=?)

sqlite> .eqp off

-- drop index
sqlite> drop index hh;

-- check again schema master table
sqlite> select * from sqlite_master where tbl_name = 'block_records';
table|block_records|block_records|4|CREATE TABLE block_records(header_hash text PRIMARY KEY, prev_hash text, height bigint,block blob, sub_epoch_summary blob, is_peak tinyint, is_block tinyint)
index|sqlite_autoindex_block_records_1|block_records|5|
index|height|block_records|15|CREATE INDEX height on block_records(height)
index|peak|block_records|16|CREATE INDEX peak on block_records(is_peak)

sqlite> .eqp on

-- the internal index is still used
sqlite> SELECT block from block_records WHERE header_hash='1c6e3fc954fed0bfd6d6d44bb850910e98fe48a086ef00e9fbe3ea7e84490b40';
QUERY PLAN
--SEARCH block_records USING INDEX sqlite_autoindex_block_records_1 (header_hash=?)
```

The indexes even had the same size when I checked, in the meantime I dropped the indexes, did a db reorg using vacuum and recreated them, so we can't use data from my side, but you could easily try on your side using:
```
.header on
sqlite> select name as index_name, sum(pgsize) as bytes from "dbstat" WHERE name in ('hh','sqlite_autoindex_block_records_1') group by name;
index_name|bytes
hh|81813504
sqlite_autoindex_block_records_1|85876736
```

Reply:
Similar topic:
In chia/full_node/coin_store.py, I saw the index coin_spent Creation statemtn was already removed, but the drop index wasn't actually enforced on the nodes.  On my main full node the index was still there and on my test full nude it was missing, most probably, because of a complete resynctest I did lately, where the index wasn't created.

The coin_record table is by far the biggest table(~10.5TiB) with about 93% of all transactions (according to full node sql logging I ran). One additional and unnecessary index can have a slight performance impact, depending on hardware. 

If you already don't create the index for new systems, I'd recommend to enforce the index drop for older ones as well.
The impact may be negligible, but you probably don't test the performance with this index any more (like your last testnet7 load tests)?

Reply:
One concern we had with dropping the index is that if the user downgrades to the previous version, on startup node will create it again, which takes some time and node is already slow on startup. So we were worried about that scenario - our idea is we would drop it after a few releases

Reply:
Sounds reasonable. 
But then you are speaking of nodes on old/slow HW with the db probably not located at SSD and in this case, I'd rather remove the index, to get rid of unnecesary index maintenance, necessary for every single DML, than to keep to reduce a one time longer startup.